### ISSUE TITLE
k8s.gcr.io/pause:3.1 failed to detect and pick up correct ppc64le architecture image variant

### ISSUE THREAD

**Issue by runlevel5** [containers/podman-compose#102]
Hi team

I would like to report an issue with `podman-compose` failing to bring up service because the `k8s.gcr.io/pause` is not the ppc64le version.

Here is my setup:

```
$ uname -r
5.4.12-200.fc31.ppc64le # Fedora 32 (Rawhide) running on RaptorCS Blackbird POWER9 8core CPU
$ podman-compose version
podman-composer version  0.1.6dev
podman --version
podman version 1.7.0
0
```

```yaml
# docker-compose.yml
version: '3'
services:
  db:
    image: ppc64le/postgres
    restart: always
    ports:
      - 5432:5432
    volumes:
      - db01:/var/lib/postgresql/data
volumes:
  db01: {}
```

```shell
[tle@localhost baghunt]$ podman-compose up
podman pod create --name=baghunt --share net -p 5432:5432
ERRO[0003] Error freeing pod lock after failed creation: no such file or directory
Error: unable to create pod: error adding Infra Container: unable to pull k8s.gcr.io/pause:3.1: unable to pull image: Image architecture mismatch: image uses "amd64", expecting "ppc64le"
125
podman volume inspect baghunt_db01 || podman volume create baghunt_db01
podman create --name=baghunt_db_1 --pod=baghunt --label io.podman.compose.config-hash=123 --label io.podman.compose.project=baghunt --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=db --mount type=bind,source=/home/tle/.local/share/containers/storage/volumes/baghunt_db01/_data,destination=/var/lib/postgresql/data,bind-propagation=z --add-host db:127.0.0.1 --add-host baghunt_db_1:127.0.0.1 ppc64le/postgres
Error: no pod with name or ID baghunt found: no such pod
125
podman start -a baghunt_db_1
Error: unable to find container baghunt_db_1: no container with name or ID baghunt_db_1 found: no such container
125
```

I was looking into a way to explicitly declare the architecture with `podman` but nothing useful in any documentation.

Any help would be greatly appreciated.

**Comment by mheon** [comment]
We're fixing this in Podman - the next release will have a resolution.

**Comment by joneslee85** [comment]
I dig in a bit into the source code, it turns out the `pause` image is hardcoded.

I learn that Google does offer other variants:

<img width="827" alt="Screen Shot 2020-01-23 at 22 50 58" src="https://user-images.githubusercontent.com/135605/72982352-f5b83b00-3e32-11ea-9a23-99b205a941c1.png">

Am I naive to think that appending the arch in the image path would solve the issue?

**Comment by mheon** [comment]
The issue is that the pause image codes an incorrect architecture for its
non-AMD64 images - they claim to be built for AMD64 when they are actually
POWER, Arm, etc. We recently added checks to verify this on the Podman
side, but have determined that this is probably a back idea - we'll be
removing the checks again, and the images will go back to working.

**Comment by joneslee85** [comment]
Related PR https://github.com/containers/libpod/issues/4917

I am going to close this PR once the check is removed.

**Comment by runlevel5** [comment]
The issue is now addressed in podman 1.8
